contracts/BC_fusion/StBNB.sol

```diff
@@ -0,0 +1,354 @@
+        return (_bnbAmount * _getTotalShares()) / _getTotalPooledBNB();
+    }
+
+    /**
+     * @return the amount of BNB that corresponds to `_sharesAmount` token shares.
+     */
+    function getPooledBNBByShares(uint256 _sharesAmount) public view returns (uint256) {
+        return (_sharesAmount * _getTotalPooledBNB()) / _getTotalShares();
+    }
+
+    /**
+     * @return the total amount (in wei) of BNB controlled by the protocol.
+     * @dev This is used for calculating tokens from shares and vice versa.
+     * @dev This function is required to be implemented in a derived contract.
+     */
+    function _getTotalPooledBNB() internal view virtual returns (uint256);
+
+    /**
+     * @notice Moves `_amount` tokens from `_sender` to `_recipient`.
+     * Emits a `Transfer` event.
+     * Emits a `TransferShares` event.
+     */
+    function _transfer(address _sender, address _recipient, uint256 _amount) internal {
+        require(_sender != address(0), "TRANSFER_FROM_ZERO_ADDR");
+        require(_recipient != address(0), "TRANSFER_TO_ZERO_ADDR");
+        require(_recipient != address(this), "TRANSFER_TO_STBNB_CONTRACT");
+
+        uint256 currentSenderShares = _shares[_sender];
+        require(currentSenderShares >= _amount, "BALANCE_EXCEEDED");
+        unchecked {
+            _shares[_sender] = currentSenderShares - _amount;
+            // Overflow not possible: the sum of all shares is capped by totalSupply, and the sum is preserved by
+            // decrementing then incrementing.
+            _shares[_recipient] += _amount;
+        }
+
+        emit Transfer(_sender, _recipient, _amount);
+    }
+
+    /**
+     * @notice Sets `_amount` as the allowance of `_spender` over the `_owner` s tokens.
+     *
+     * Emits an `Approval` event.
+     *
+     * NB: the method can be invoked even if the protocol paused.
+     *
+     * Requirements:
+     *
+     * - `_owner` cannot be the zero address.
+     * - `_spender` cannot be the zero address.
+     */
+    function _approve(address _owner, address _spender, uint256 _amount) internal {
+        require(_owner != address(0), "APPROVE_FROM_ZERO_ADDR");
+        require(_spender != address(0), "APPROVE_TO_ZERO_ADDR");
+
+        _allowances[_owner][_spender] = _amount;
+        emit Approval(_owner, _spender, _amount);
+    }
+
+    /**
+     * @dev Updates `owner` s allowance for `spender` based on spent `amount`.
+     *
+     * Does not update the allowance amount in case of infinite allowance.
+     * Revert if not enough allowance is available.
+     *
+     * Might emit an {Approval} event.
+     */
+    function _spendAllowance(address _owner, address _spender, uint256 _amount) internal {
+        uint256 currentAllowance = allowance(_owner, _spender);
+        if (currentAllowance != INFINITE_ALLOWANCE) {
+            require(currentAllowance >= _amount, "ALLOWANCE_EXCEEDED");
+            unchecked {
```

Comment:
As you can see `_approve` has some `require` condition, will the `unchecked` here jump those condition if `_approve` revert?

Reply:
No